app/forms/concerns/science_gcse_helper.rb

```diff
@@ -0,0 +1,177 @@
+      self.grade = 'A*A' and return if sanitize(grade) == 'AA*'
+      grade.reverse! and return if DOUBLE_GCSE_GRADES.include?(sanitize(grade).reverse)
+
+      errors.add(:double_award_grade, :invalid)
+    end
+  end
+
+  def triple_award_grade_format
+    return unless triple_award?
+
+    grade_hash = {
+      biology_grade:,
+      chemistry_grade:,
+      physics_grade:,
+    }
+
+    grade_hash.each do |key, grade|
+      next if grade.blank?
+
+      public_send("#{key}=", grade)
+      errors.add(key, :invalid) unless SINGLE_GCSE_GRADES.include?(sanitize(grade))
+    end
+  end
+
+  def grade_length
+    errors.add(:grade, :blank) if grade.blank? && science?
+    errors.add(:single_award_grade, :blank) if grade.blank? && single_award?
+    errors.add(:double_award_grade, :blank) if grade.blank? && double_award?
+    errors.add(:biology_grade, :blank) if biology_grade.blank? && triple_award?
+    errors.add(:chemistry_grade, :blank) if chemistry_grade.blank? && triple_award?
+    errors.add(:physics_grade, :blank) if physics_grade.blank? && triple_award?
+  end
+
+  def invalid_grades
+    {
+      gcse: /[^1-9A-GU*\s-]/i,
+      scottish_national_5: /[^A-D1-7\s-]/i,
+    }
+  end
+
+  def log_validation_errors(field)
+    return unless errors.key?(field)
+
+    error_message = {
+      field: field.to_s,
+      error_messages: errors[field].join(' - '),
+      value: grade || constituent_grades,
+    }
+
+    Rails.logger.info("Validation error: #{error_message.inspect}")
+  end
+
+  def set_grade
+    return if triple_award?
+
+    case grade
+    when 'other'
+      other_grade
+    when 'not_applicable'
+      'N/A'
+    when 'unknown'
+      'Unknown'
+    else
+      sanitize(grade)
+    end
+  end
+
+  def set_triple_award_grades
+    if triple_award?
+      {
+        biology: grade_hash(biology_grade),
+        physics: grade_hash(physics_grade),
+        chemistry: grade_hash(chemistry_grade),
+      }
+    end
+  end
+
+  def grade_hash(grade)
+    {
+      grade: sanitize(grade),
+    }
+  end
+
+  def sanitize(grade)
+    if ALL_GCSE_GRADES.exclude?(grade) && grade_contains_two_numbers?(grade)
+      remove_special_characters_and_add_dash_between_numbers(grade)
+    elsif DOUBLE_GCSE_GRADES.exclude?(grade)
+      remove_special_characters_and_upcase(grade)
+    else
+      grade
+    end
+  end
+
+  def new_record?
+    qualification.nil?
+  end
+
+  def grade_is_other?
+    grade == 'other'
+  end
+
+  def triple_award?
+    subject == ApplicationQualification::SCIENCE_TRIPLE_AWARD
+  end
+
+  def double_award?
+    subject == ApplicationQualification::SCIENCE_DOUBLE_AWARD
+  end
+
+  def single_award?
+    subject == ApplicationQualification::SCIENCE_SINGLE_AWARD
+  end
+
+  def science?
+    subject == ApplicationQualification::SCIENCE
+  end
+
+  def gsce_qualification_type?
+    qualification.qualification_type == 'gcse'
+  end
+  alias gcse? gsce_qualification_type?
```

Comment:
Why do you need this alias?

Reply:
gsce_qualification_type? is used in candidate interface and I prefer to use gcse? in support (that's the name of the option)

Reply:
Ok, then you should be able to remove [this method](https://github.com/DFE-Digital/apply-for-teacher-training/pull/8749/files#diff-6911382f09dcdbb528cdd81044ce846fc528ea0c2d5add24427ee32669630d40R185). And you got a typo `gsce_` -> `gcse_`

Reply:
Done! The problem with the typo is that the typo is on candidate interface and I am hesitant to change candidate interface at this point tbh.